apps/api/src/coscrad-cli/seed-test-data-with-command.cli-command.ts

```diff
@@ -0,0 +1,147 @@
+import { Ack, CommandHandlerService } from '@coscrad/commands';
+import { isNonEmptyString, isUndefined } from '@coscrad/validation-constraints';
+import { Inject } from '@nestjs/common';
+import { ID_MANAGER_TOKEN, IIdManager } from '../domain/interfaces/id-manager.interface';
+import { AggregateType } from '../domain/types/AggregateType';
+import { InternalError } from '../lib/errors/InternalError';
+import { CliCommand, CliCommandOption, CliCommandRunner } from './cli-command.decorator';
+import { COSCRAD_LOGGER_TOKEN, ICoscradLogger } from './logging';
+
+interface SeedTestDataWithCommandOptions {
+    type: string;
+    payloadOverrides?: Record<string, unknown>;
+}
+
+/**
+ * TODO Move this to the test data directory. We may want to inject a `TestDataManager`
+ * service that can find these for us.
+ */
+const allFsas = [
+    {
+        type: 'CREATE_SONG',
+        payload: {
+            aggregateCompositeIdentifier: { id: 'dummy-song', type: AggregateType.song },
+            title: 'test-song-name (language)',
+            titleEnglish: 'test-song-name (English)',
+            lyrics: 'la la la',
+            audioURL: 'https://www.mysound.org/song.mp3',
+        },
+    },
+];
+
+@CliCommand({
+    name: 'seed-test-data-with-command',
+    description: 'executes a fixture command (by type) with optional overrides to seed test data',
+})
+export class SeedTestDataWithCommand extends CliCommandRunner {
+    constructor(
+        private readonly commandHandlerService: CommandHandlerService,
+        @Inject(ID_MANAGER_TOKEN) private readonly idManager: IIdManager,
+        @Inject(COSCRAD_LOGGER_TOKEN) private readonly logger: ICoscradLogger
+    ) {
+        super();
+    }
+
+    async run(_passedParams: string[], options: SeedTestDataWithCommandOptions): Promise<void> {
+        const { type: commandType, payloadOverrides } = options;
+
+        const fixtureFsaSearchResult = allFsas.find(({ type }) => type === commandType);
+
+        if (isUndefined(fixtureFsaSearchResult)) {
+            this.logger.log(`failed to find a fixture command of type: ${commandType}. Exiting.`);
+
+            throw new InternalError(`No fixture command of type: ${commandType} found`);
+        }
+
+        const { payload: defaultPayload } = fixtureFsaSearchResult;
+
+        const fsaToWithOverrides = {
+            type: commandType,
+            payload: {
+                ...defaultPayload,
+                ...(payloadOverrides || {}),
+            },
+        };
+
+        this.logger.log(`attempting to execute FSA: ${JSON.stringify(fsaToWithOverrides)}`);
+
+        /**
+         * TODO We need to do this conditionally, only for create commands. For
```

Comment:
Perhaps the default logic for the `execute` should be to generate an id and append it to `aggregateCompositeIdentifier.id` if this property is missing and the command is a create command.

Reply:
This might help simplify the front-end in cases where the ID doesn't need to be known in advance.